pyrealm_build_data/splash/splash_py_version/splash_data.py

```diff
@@ -0,0 +1,834 @@
+                  and create the output file
+        Depends:  writeout
+        """
+        self.output_file = d + "splash_data_out.csv"
+        header = "date,sf,tair,pn\n"
+        if os.path.isfile(self.output_file):
+            try:
+                user_resp = raw_input("File already exists, overwrite (y/n)? ")
+            except NameError:
+                # Maybe using Python 3, try again:
+                try:
+                    user_resp = input("File already exists, overwrite (y/n)? ")
+                except:
+                    logging.exception("Could not get user input")
+                    raise
+            if user_resp.lower() == "y":
+                self.writeout(self.output_file, header)
+        else:
+            self.writeout(self.output_file, header)
+            logging.info("creating output file %s", self.output_file)
+
+    def set_sf_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_sf_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for fractional sunshine hours
+                  (e.g., 'cru' or 'watch')
+        """
+        logging.info("setting sunshine fraction preference to '%s'", n)
+        self.data_src["sf"] = n
+
+    def set_pn_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_pn_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for precipitation (e.g., 'cru'
+                  or 'watch')
+        """
+        logging.info("setting precipitation preference to '%s'", n)
+        self.data_src["pn"] = n
+
+    def set_tair_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_tair_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for air temperature (e.g., 'cru'
+                  or 'watch')
+        """
+        logging.info("setting temperature preference to '%s'", n)
+        self.data_src["tair"] = n
+
+    def add_one_month(self, dt0):
+        """
+        Name:     SPLASH_DATA.add_one_month
+        Input:    datetime date
+        Output:   datetime date
+        Features: Adds one month to datetime
+        Ref:      A. Balogh (2010), ActiveState Code
+                  http://code.activestate.com/recipes/577274-subtract-or-add-a-
+                  month-to-a-datetimedate-or-datet/
+        """
+        dt1 = dt0.replace(day=1)
+        dt2 = dt1 + datetime.timedelta(days=32)
+        dt3 = dt2.replace(day=1)
+        return dt3
+
+    def density_h2o(self, tc, p):
+        """
+        Name:     SPLASH_DATA.density_h2o
+        Input:    - float, air temperature (tc), degrees C
+                  - float, atmospheric pressure (p), Pa
+        Output:   float, density of water, kg/m^3
+        Features: Calculates density of water at a given temperature and
+                  pressure
+        Ref:      Chen et al. (1977)
+        """
+        # Calculate density at 1 atm:
+        po = (
```

Comment:
Number of multiplications can be reduced using `Horner's method`. Basically taking out common `tc` at every step.

po = (
    0.99983952
    + tc * (
        6.788260e-5
        + tc * (
            -9.08659e-6
            + tc * (
                1.022130e-7
                + tc * (
                    -1.35439e-9
                    + tc * (
                        1.471150e-11
                        + tc * (
                            -1.11663e-13
                            + tc * (
                                5.044070e-16
                                - tc * (
                                    1.00659e-18 * tc
                                )
                            )
                        )
                    )
                )
            )
        )
    )
)

Reply:
It absolutely can - but this is one of the 'original' implementation files where I want to leave the process untouched.